src/Explorer/Web/ContractView.hs

```diff
@@ -5,36 +5,62 @@ module Explorer.Web.ContractView
   (ContractView(..), contractView)
   where
 
+import Control.Monad (forM_)
 import Control.Newtype.Generics (op)
 import Language.Marlowe.Pretty ( pretty )
-import Language.Marlowe.Runtime.Types.ContractJSON (ContractJSON(..), getContractJSON)
+import Language.Marlowe.Runtime.Types.ContractJSON
+  ( ContractJSON(..), getContractJSON
+  , Transaction(..), Transactions(..), getContractTransactions
+  )
 import Language.Marlowe.Semantics.Types (Contract, State)
 import qualified Language.Marlowe.Runtime.Types.ContractJSON as CJ
 import Prelude hiding ( head )
-import Text.Blaze.Html5 ( b, string, Html, ToMarkup(toMarkup), (!), Markup, code )
+import Text.Blaze.Html5 ( Html, Markup, ToMarkup(toMarkup), (!), b, code, p, string, toHtml )
 import Text.Blaze.Html5.Attributes ( style )
 import Text.Printf ( printf )
 
 import Explorer.Web.Util
 import Opts (Options (optRuntimeHost, optRuntimePort), RuntimeHost (..), RuntimePort (..))
 
+
 contractView :: Options -> Maybe String -> Maybe String -> IO ContractView
-contractView _    _   Nothing    = return $ ContractViewError "Need to specify a contractId"
-contractView opts tab (Just cid) = do
+
+contractView opts tab@(Just "txs") (Just cid) = do
+  let
+    rhost = op RuntimeHost . optRuntimeHost $ opts
+    rport = op RuntimePort . optRuntimePort $ opts
+    urlPrefix = printf "http://%s:%d/" rhost rport
+  cjs <- getContractJSON urlPrefix cid
+  case cjs of
+    Left str -> pure $ ContractViewError str
+    Right cjson -> do
+      let link = CJ.transactions . CJ.links $ cjson
+      etx <- getContractTransactions urlPrefix link
+      pure $ case etx of
+        Left str -> ContractViewError str
+        Right tx -> extractInfo (parseTab tab) cjson (Just tx)
+
+contractView opts tab@(Just _) (Just cid) = do
   let
     rhost = op RuntimeHost . optRuntimeHost $ opts
     rport = op RuntimePort . optRuntimePort $ opts
-  v <- getContractJSON (printf "http://%s:%d/" rhost rport) cid
-  return $ case v of
+  cjs <- getContractJSON (printf "http://%s:%d/" rhost rport) cid
+  return $ case cjs of
     Left str -> ContractViewError str
-    Right cjson -> extractInfo (parseTab tab) cjson
+    Right cjson -> extractInfo (parseTab tab) cjson Nothing
+
+contractView _opts _tab _cid = return $ ContractViewError "Need to specify a contractId"
```

Comment:
```suggestion
contractView _opts _tab Nothing = return $ ContractViewError "Need to specify a contractId"
```
I think it is better to specify what we know, that way if there is a missing option the compiler will tell us